README.md

```diff
@@ -17,25 +18,267 @@ go get github.com/EdgeCast/ec-sdk-go
+from around the world.
+
+For more information, please read the [official documentation](https://docs.whitecdn.com/cdn/index.html#Route/Route_DNS.htm%3FTocPath%3DRoute%2520(DNS)%7C_____0)
+
+To use the Route(DNS) service, use the API Token provided to you.
+
+#### Master Server Group
+
+A master server group allows quick and easy management of master name servers, 
+while a secondary zone group defines the secondary zones that will be imported 
+from servers defined in a master server group and any TSIG keys that should be 
+used for the zone transfer.
+
+```
+import (
+	"github.com/EdgeCast/ec-sdk-go/edgecast"
+	"github.com/EdgeCast/ec-sdk-go/edgecast/routedns"
+)
+// ...
+	sdkConfig := edgecast.NewSDKConfig()
+	sdkConfig.APIToken = "MY API TOKEN"
+	routeDNSService, err := routedns.New(sdkConfig)
+	masterServerGroup := buildMasterServerGroup()
+	addParams := routedns.NewAddMasterServerGroupParams()
+	addParams.MasterServerGroup = masterServerGroup
+	// ...
+	resp, err := routeDNSService.AddMasterServerGroup(*addParams)
+// ...
+}
+```
+
+### Real Time Log Delivery (RTLD)
+
+Our Real-Time Log Delivery (RTLD) servicee delivers log data in near real-time 
+to a variety of destinations.
+
+For more information, please read the [official documentation](https://docs.edgecast.com/cdn/index.html#RTLD/RTLD.htm%3FTocPath%3DReports%2520and%2520Log%2520Data%7CReal-Time%2520Log%2520Delivery%7C_____0).
+
+To use the Rules Engine service, use OAuth 2.0 Credentials.
+
+#### Real-Time Log Delivery Rate Limiting (RTLD Rate Limiting)
+
+Delivers log data that describes requests for which Web Application Firewall (WAF) 
+enforced a rate limit as defined through a rate rule.
+
+```
+import (
+	"github.com/EdgeCast/ec-sdk-go/edgecast"
+	"github.com/EdgeCast/ec-sdk-go/edgecast/waf"
+)
+// ...
+	sdkConfig := edgecast.NewSDKConfig()
+	sdkConfig.IDSCredentials = "IDS CREDENTIALS"
+	rtldService, err := rtld.New(sdkConfig)
+	addParams := profiles_rl.NewProfilesRateLimitingAddCustomerSettingParams()
+	addParams.SettingDto = &rtldmodels.RateLimitingProfileDto{
+		// ...
+	}
+	addResp, err :=
+		rtldService.ProfilesRl.ProfilesRateLimitingAddCustomerSetting(addParams)
+// ...
+}
+```
+
+### Rules Engine
+
+Our Rules Engine allows the customization of requests handled by our CDN. 
+Sample customizations that may be performed are:
+
+Override or define a custom cache policy
+Secure or deny requests for sensitive content
+Redirect requests to a different URL
+
+For more information, please read the [official documentation](https://docs.whitecdn.com/cdn/index.html#HRE/HRE.htm).
+
+To use the Rules Engine service, use OAuth 2.0 Credentials.
+A Policy should be constructed as a JSON object passed as a string.
+
+```
+import (
+	"github.com/EdgeCast/ec-sdk-go/edgecast"
+	"github.com/EdgeCast/ec-sdk-go/edgecast/waf"
+)
+// ...
+	policyString := `{
+		// ...
+	}`
+	sdkConfig := edgecast.NewSDKConfig()
+	sdkConfig.IDSCredentials = "IDS CREDENTIALS"
```

Comment:
The IDS credentials have to be set like this: 
```
idsCredentials := edgecast.IDSCredentials{
		ClientID:     clientID,
		ClientSecret: clientSecret,
		Scope:        scope,
	}
```

Reply:
updated